### PR TITLE
Add BigQuery configuration doc

### PR DESCRIPTION
**Pull request by KasunAratthanage**

### CODE REVIEW

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Setting up the BigQuery Environment  
+
+The BigQuery connector allows you to access the [BigQuery REST API](https://cloud.google.com/bigquery/docs/reference/rest) through WSO2 through WSO2 EI.
+
+To work with the BigQuery connector, you need to have a Google Cloud Platform account. If you do not have a Google Cloud Platform account, go to [console.cloud.google.com](https://console.cloud.google.com/freetrial), and create a Google Cloud Platform trial account.
+
+BigQuery uses the OAuth 2.0 protocol for authentication and authorization. All requests to the BigQuery REST API must be authorized by an authenticated user. For information on how to obtain authentication and authorization user credentials, you can get the user credentials from the Google Cloud Platfrom using two different manner. See the following section.
```
Comment (abeykoon) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:7]: OAuth 2.0 is for authorization, not for authentication.

Comment (abeykoon) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:7]: All requests to the BigQuery REST API will be authorized against a registered user. 
Developers can generate user credentials from the Google Cloud Platform using two different mechanisms. 
See the following sections for detail.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+10. Get the authorization code by sending a GET request to the following URL. Replace the `<redirect_uri>` and `<client_ID>` with the redirect uri and client ID values noted in the previous steps. Enter the following URL in your web browser.
+ 
+    ```
+    https://accounts.google.com/o/oauth2/auth?redirect_uri=<redirect_uri>&response_type=code&client_id=<client_ID>&scope=https://www.googleapis.com/auth/bigquery&approval_prompt=force&access_type=offline
+    ```
+    Note the authorization code for future use.
+    
+    <img src="../../../../assets/img/connectors/bigquery-get-authorization-code.png" title="Get authorization code" width="600" alt="Get authorization code"/> 
+    
+11. Get the `access token` and `refresh token` by sending a POST request to the URL given below. Be sure to use an **x-www-form-urlencoded** body with the `<authorization_code>`, `<client_id>`, `<client_secret>`, and `<redirect_uri>` values noted before, and also set the `grant_type` to **authorization_code**. Note the access token and refresh token for future use.
```
Comment (abeykoon) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:56]: You will need them to configure WSO2 EI Big Query Connector.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 1 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:7]: Fixed.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 2 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:56]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
+# Setting up the BigQuery Environment  
+
+The BigQuery connector allows you to access the [BigQuery REST API](https://cloud.google.com/bigquery/docs/reference/rest) through WSO2 through WSO2 EI.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:3]:
```suggestion
The BigQuery connector allows you to access the [BigQuery REST API](https://cloud.google.com/bigquery/docs/reference/rest) through WSO2 EI.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    ```
+    <img src="../../../../assets/img/connectors/bigquery-gettoken-postman.png" title="Bigquery get token using postman" width="800" alt="Bigquery get token using postman"/> 
+
+### Obtaining credentials using the service account
+
+1. Open the [Service Accounts](https://console.cloud.google.com/projectselector2/iam-admin/serviceaccounts) page in the GCP console.
+
+    <img src="../../../../assets/img/connectors/bigquery-service-account.png" title="Bigquery service account" width="600" alt="Bigquery service account"/> 
+
+2. Select your project and click Open.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:69]: Let's bold all UX labels.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 3 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:3]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 4 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:69]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+    <img src="../../../../assets/img/connectors/bigquery-create-dataset1.png" title="Bigquery create Dataset step1" width="600" alt="Bigquery create Dataset step1"/> 
+
+2. Enter required Dataset details and click **Create Dataset**.
+
+    <img src="../../../../assets/img/connectors/bigquery-create-dataset2.png" title="Bigquery create Dataset step2" width="600" alt="Bigquery create Dataset step2"/> 
+
+**Creating Table**
+
+1. After creating the Dataset, click created **Dataset**. You can see the following details. Then Click **Create Table**.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:127]:
```suggestion
1. After creating the Dataset, click the created **Dataset**. You can see the following details. Then click **Create Table**. 
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+    <img src="../../../../assets/img/connectors/bigquery-create-project2.png" title="Bigquery create project step2" width="600" alt="Bigquery create project step2"/> 
+
+3.  Enter new project details.
+
+    <img src="../../../../assets/img/connectors/bigquery-create-project3.png" title="Bigquery create project step3" width="600" alt="Bigquery create project step3"/> 
+
+**Creating Dataset**
+
+1. After creating the Project, click created **project**. You can see the following details. Then Click Create **Dataset**.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:117]:
```suggestion
1. After creating the Project, click the created **project**. You can see the following details. Then click Create **Dataset**.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+    <img src="../../../../assets/img/connectors/bigquery-create-key.png" title="Bigquery service account create key" width="600" alt="Bigquery service account create key"/> 
+
+9. Select the key type as **P12** and click **Create**. Then created key will be downloaded.
+
+### Creating Project, Dataset and Table
+
+**Creating Project**
+
+1. Open the Bigquery console.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:101]:
```suggestion
1. Open the BigQuery console.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+7. Go to the service account for which you wish to create a key, **click** created Service account in that row.
+
+    <img src="../../../../assets/img/connectors/bigquery-select-created-service-account.png" title="Bigquery enter service account grant user access" width="600" alt="Bigquery enter service account grant user access"/> 
+
+8. Click **Create key**.
+
+    <img src="../../../../assets/img/connectors/bigquery-create-key.png" title="Bigquery service account create key" width="600" alt="Bigquery service account create key"/> 
+
+9. Select the key type as **P12** and click **Create**. Then created key will be downloaded.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:95]:
```suggestion
9. Select the key type as **P12** and click **Create**. Then the created key will be downloaded.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+5. Select a **role** you wish to grant to the service account, and then click **Continue**.
+
+    <img src="../../../../assets/img/connectors/bigquery-service-account-role.png" title="Bigquery enter service account role" width="600" alt="Bigquery enter service account role"/> 
+
+6. Grant users access to this service account (optional), and click **Done**.
+
+    <img src="../../../../assets/img/connectors/bigquery-service-account-grant-user-access.png" title="Bigquery enter service account grant user access" width="600" alt="Bigquery enter service account grant user access"/> 
+
+7. Go to the service account for which you wish to create a key, **click** created Service account in that row.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:87]:
```suggestion
7. Go to the service account for which you wish to create a key and **click** the created Service account in that row.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+4. Enter **Service account details**, and then click **Create**.
+
+    <img src="../../../../assets/img/connectors/bigquery-enter-service-account-details.png" title="Bigquery enter service account" width="600" alt="Bigquery enter service account"/> 
+
+5. Select a **role** you wish to grant to the service account, and then click **Continue**.
+
+    <img src="../../../../assets/img/connectors/bigquery-service-account-role.png" title="Bigquery enter service account role" width="600" alt="Bigquery enter service account role"/> 
+
+6. Grant users access to this service account (optional), and click **Done**.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:83]:
```suggestion
6. Grant users access to this service account (optional) and click **Done**.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+3. Click **Create Service Account**.
+  
+    <img src="../../../../assets/img/connectors/bigquery-create-service-account.png" title="Bigquery create service account" width="600" alt="Bigquery create service account"/> 
+
+4. Enter **Service account details**, and then click **Create**.
+
+    <img src="../../../../assets/img/connectors/bigquery-enter-service-account-details.png" title="Bigquery enter service account" width="600" alt="Bigquery enter service account"/> 
+
+5. Select a **role** you wish to grant to the service account, and then click **Continue**.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:79]:
```suggestion
5. Select a **role** you wish to grant to the service account and click **Continue**.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+**Obtaining a client ID and client secret**
+
+1. Go to [https://accounts.google.com/SignUphttps://accounts.google.com/SignUp](https://accounts.google.com/SignUp) and create a Google account.
+
+2. Go to [https://console.developers.google.com/projectselector/apis/credentials](https://console.developers.google.com/apis/credentials), and sign in to your **Google account**.
+    
+    <img src="../../../../assets/img/connectors/bigquery-credentials-page.png" title="Bigquery-credentials-page" width="600" alt="Bigquery-credentials-page"/>   
+   
+3. If you do not already have a project, you create a new project, click **Create credentials** and then select **OAuth client ID**.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:21]:
```suggestion
3. If you do not already have a project, you can create a new project. Click **Create credentials** and then select **OAuth client ID**.     
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+   
+3. If you do not already have a project, you create a new project, click **Create credentials** and then select **OAuth client ID**.     
+   
+    <img src="../../../../assets/img/connectors/bigquery-create-credentials.png" title="Select OAuth client ID" width="600" alt="Select OAuth client ID"/> 
+   
+4. Next, **select** Web Application, and **create a client**.
+  
+    <img src="../../../../assets/img/connectors/bigquery-select-web-application.png" title="Select web application" width="600" alt="Select web application"/> 
+
+5. Add [https://www.google.com](https://www.google.com) as the redirect URI (You can add any URI which you need as redirect URI) under **Authorized redirect URIs**, and then click **Create**. This displays the **client ID** and **client secret**.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:29]:
```suggestion
5. Add [https://www.google.com](https://www.google.com) as the redirect URI (you can add any URI that you need as redirect URI) under **Authorized redirect URIs**, and then click **Create**. This displays the **client ID** and **client secret**.
```

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 5 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:127]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 6 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:117]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 7 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:101]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 8 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:95]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 9 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:87]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 10 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:83]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 11 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:79]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 12 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:21]: Updated.

File: en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md (hunk 13 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/bigquery-connector/bigquery-connector-configuration.md:29]: Updated.

**Pull request merged**